src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java

```diff
@@ -0,0 +1,231 @@
+    }
+
+    private CallbackResponse checkSubmitError(CallbackParams callbackParams,
+                                              CaseData caseData,
+                                              LocalDateTime dateTime) {
+        // Show error message if defendant tries to submit response again
+        if (hasDefendantAlreadySubmitted(callbackParams, caseData)) {
+            return createErrorResponse(ERROR_DEFENDANT_RESPONSE_SUBMITTED);
+        }
+
+        // Show error message if defendant tries to submit a response after deadline has passed
+        if (isResponseSubmissionPassedDeadline(callbackParams, caseData, dateTime)) {
+            return createErrorResponse("You cannot submit a response now as you have passed your deadline");
+        }
+
+        return null;
+    }
+
+    private YesOrNo findOutIsRespondent1(CallbackParams callbackParams) {
+        YesOrNo isRespondent1 = YES;
+        if (solicitorRepresentsOnlyOneOrBothRespondents(callbackParams, RESPONDENTSOLICITORTWO)) {
+            // 1V2 Different Solicitors + Respondent 2 only
+            isRespondent1 = NO;
+        }
+        return isRespondent1;
+    }
+
+    private CallbackResponse createErrorResponse(String errorMessage) {
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .errors(List.of(errorMessage))
+            .build();
+    }
+
+    private CaseData.CaseDataBuilder updateCaseData(CaseData caseData, YesOrNo isRespondent1, RequestedCourt requestedCourt1) {
+        var updatedCaseData = caseData.toBuilder()
+            .respondent1Copy(caseData.getRespondent1())
+            .isRespondent1(isRespondent1)
+            .respondent1DQ(Respondent1DQ.builder()
+                               .respondent1DQRequestedCourt(
+                                   requestedCourt1)
+                               .build());
+
+        if (caseData.getRespondent2() != null) {
+            updatedCaseData.respondent2DQ(
+                Respondent2DQ.builder()
+                    .respondent2DQRequestedCourt(requestedCourt1).build());
+        }
+
+        updatedCaseData.respondent1DetailsForClaimDetailsTab(updatedCaseData.build().getRespondent1()
+                                                                 .toBuilder().flags(null).build());
+
+        if (ofNullable(caseData.getRespondent2()).isPresent()) {
+            updatedCaseData
+                .respondent2Copy(caseData.getRespondent2())
+                .respondent2DetailsForClaimDetailsTab(updatedCaseData.build().getRespondent2()
+                                                          .toBuilder().flags(null).build());
+        }
+        return updatedCaseData;
+    }
+
+    private RequestedCourt createRequestedCourt(List<LocationRefData> locations, CaseData caseData) {
+        DynamicList courtLocationList = courtLocationUtils.getLocationsFromList(locations);
+        RequestedCourt.RequestedCourtBuilder requestedCourt1 = RequestedCourt.builder();
```

Comment:
is there any reason for calling it requestedCourt1 if apparently there isn't a second one (requestedCourt2)?